test/DeployBase.t.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.23;
+
+import {Test} from "forge-std/Test.sol";
+import {IEarningPowerCalculator} from "../src/interfaces/IEarningPowerCalculator.sol";
+import {MintRewardNotifier} from "../src/notifiers/MintRewardNotifier.sol";
+import {TransferRewardNotifier} from "../src/notifiers/TransferRewardNotifier.sol";
+import {Staker} from "../src/Staker.sol";
+import {DeployBaseFake} from "./fakes/DeployBaseFake.sol";
+import {DeployMultipleRewardNotifiersFake} from "./fakes/DeployMultipleRewardNotifiersFake.sol";
+import {ERC20Fake} from "./fakes/ERC20Fake.sol";
+import {ERC20VotesMock} from "./mocks/MockERC20Votes.sol";
+
+contract DeployBaseTest is Test {
+  ERC20Fake rewardToken;
+  ERC20VotesMock govToken;
+  DeployBaseFake deployScript;
+
+  function setUp() public {
+    rewardToken = new ERC20Fake();
+    vm.label(address(rewardToken), "Reward Token");
+
+    govToken = new ERC20VotesMock();
+    vm.label(address(govToken), "Governance Token");
+
+    deployScript = new DeployBaseFake(rewardToken, govToken);
+  }
+}
+
+contract Run is DeployBaseTest {
+  function test_StakingSystemDeploy() public {
```

Comment:
Wondering out loud: should we follow scopelint spec naming conventions for the script tests? It looks like we did by implementing the tests inside a contract named `Run`. If so, we should probably make the tests names sentences that describe the intended behavior.

Reply:
I could go either way. These tests will not show up in scopelint as run isn't defined on the tested contracts.

Reply:
Yeah, I could go either way as well. If they won't show up, then we might as well just put the tests under one contract, or at least name them a little bit better. They don't have to be under method-named contracts. I do think the test names could also be better, even if they're not showing up. Explain what is being tested rather than just being a noun.